### DISCUSSION TITLE
[@types/react] React 18 issues with reactNode

### DISCUSSION THREAD

**Discussion by rajzik** [DefinitelyTyped/DefinitelyTyped#59841] [Issues with a @types package] [answered]
Seems like we have multiple issues with react@18 types that are breaking even though they shouldn't be.

## `ReactNode` issue

```tsx
import { Typography } from '@mui/material';

const foo = 22;
const myDiv = <div>my {foo}</div>; // No error
const myTypography = <Typography>my {foo}</Typography>; // This JSX tag's 'children' prop expects a single child of type 'ReactNode', but multiple children were provided.ts(2746)
```

### Typography props
![image](https://user-images.githubusercontent.com/10364836/162897209-a27d22ed-e6e3-4dc5-bbbc-5582c0e90069.png)

### div props

![image](https://user-images.githubusercontent.com/10364836/162897301-87495aa2-5ba6-4526-ab34-5daaaa61c519.png)

## env info

```
- @types/react@18.0.2
- typescript@4.6.3
```

I am not really sure if this is issue with react types or mui.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
I [cannot reproduce any issue with the posted code](https://www.typescriptlang.org/play?jsx=5#code/JYWwDg9gTgLgBAbzgFQJ6QOZQIZgBapwC+cAZlBCHAOQACIArsAPQjYwCmUw2ANtQG4AUEIDGEAHYBneKQgQ4AXjgAmFcPHT4IVABFgANyVwAPABNDAPh2I5EIieYWDlgXGbM4AOQVcKUMUkZOB00TBx8QmUTMIgsXAJrQgQ7B2ZY+MjXd09kPGApOAApAGUADTgYbAxqQupRfN4zKA4JajgwCjA4DgAPMA5RGELsOClgCQxeDjgG4Ca4CFJK9BnqACUObCGfMw5qABo4ACMGbQZeGGAwadnG5ta4AHcuGc6IA2A9swA6YYAKFQAdgALAA2ACUQA)

Make sure you have a single version of `@types/react` in your dependency tree. For more info, check out https://github.com/facebook/react/issues/24304#issuecomment-1094565891

**Reply by rajzik** [reply]
I have resolutions in my package json and my yarn lock seems like it has only one @types/react

```json
  "resolutions": {
    "react-docgen-typescript": "^2.1.1",
    "@types/react": "^18.0.2",
    "@types/react-dom": "^18.0.0"
  },
```

I can only replicate it in our design system monorepo that is based on mui and runs in storybook.

I guess we can mark this as answered because I cannot replicate it elsewere.

**Comment by hellatan** [comment]
@rajzik try removing the fields in `resolutions` and see if `yarn dedupe` works. That solved my issues. https://stackoverflow.com/a/71845991/339698
